enhancements/hide_container_mountpoints.md

```diff
@@ -0,0 +1,139 @@
+utilization of systemd and overall overhead of the host OS running OpenShift.
+Changing systemd to reduce its scanning overhead is tracked in [BZ
+1819868](https://bugzilla.redhat.com/show_bug.cgi?id=1819868), but we can work
+around this exclusively within openshift. Using a separate mountpoint to for
+both crio-o and kubelet mounts can completely segregate all container-specific
+mounts away from any systemd interaction whatsoever.
+
+### Goals
+
+- cri-o places all container mountpoints in a separate mount namespace from systemd
+- kubelet places all container mountpoints in this same separate mount namespace from systemd
+- systemd does not see any cri-o or kubelet mountpoints
+- cri-o containers still see all appropriate kubelet mountpoints
+- Restarting either crio.service or kubelet.service does not result in the namespaces getting out-of-sync
+
+### Non-Goals
+
+- Fix systemd mountpoint scanning
+- Change cri-o pinns to support pinning mount namespaces
+
+## Proposal
+
+We can create a separate mount namespace and cause both crio and kubelet to
+launch within it to hide their many many mounts from systemd by creating:
+
+- A service called container-mount-namespace.service which spawns a separate
+  namespace (via systemd's 'PrivateMounts' mechanism), and then sleeps forever.
+  We don't want to create the namespace in crio.service or kubelet.service,
+  since if either one restarts they would lose eachother's namespaces.
+
+- An override file for each of crio.service and kubelet.service which wrap the
+  original command under 'nsenter' so they both use the mount namespace created
+  by 'container-mount-namespace.service'
+
+With these in place, both kubelet and crio create their mounts in the new
+shared (with eachother) but private (from systemd) namespace.
+
+### User Stories
+
+The end-user experience should not be affected in any way by this proposal, as
+there is no outward API changes. There is some supportability change though,
+since anyone attempting to inspect the cri-o or kubelet mountpoints externally
+would need to be aware that these are now available in a different namespace
+than the default top-level systemd mount namespace.
+
+### Implementation Details/Notes/Constraints
+
+Here is an example of the container-mount-namespace.service:
+
+    [Unit]
+    Description=Manages a mount namespace that both kubelet and crio can use to share their container-specific mounts
+    
+    [Service]
+    PrivateMounts=on
+    ExecStart=bash -c 'while :; do sleep infinity; done'
+
+This needs to be managed separately from either crio or kubelet to avoid the
+namespaces getting out of sync if either of those services restart. The systemd
+'PrivateMounts=on' facility does a nice job of creating a separate mountpoint,
+and both crio and kubelet can find the associated namespace via 'nsenter' as
+follows:
+
+    nsenter -m -t $(systemctl show --property MainPID --value container-mount-namespace.service) $ORIGINAL_EXECSTART
```

Comment:
both of the service files will have to also have
`After=container-mount-namespace.service`

Reply:
Probably `Requires=` too?

Reply:
Added a paragraph about this as well.